### ISSUE TITLE
Spurious analyzer diagnostic about a Null value when using a function expression in a const context

### ISSUE THREAD

**Issue by natebosch** [dart-lang/sdk#47603]
```dart
class C {
  final void Function() c;
  const C(this.c);
}

void main() {
  const C(() {});
}
```

Analyzer will report 2 errors, one of which is expected, and one which is distracting and confusing.

Expected error:
```
Arguments of a constant creation must be constant expressions.
```

Unexpected error:

```
A value of type 'Null' can't be assigned to a parameter of type 'void Function()' in a const constructor.
```

**Comment by srawlins** [comment]
Weird, still an issue.

**Comment by kallentu** [comment]
Fixed after ba00516.
